chore: include also OpenApi in destination docs (#418)
SAP/cloud-sdk: docs/js/features/connectivity/destination.mdx

The SAP Cloud SDK helps you to authenticate against the destination service and retrieve destinations into your application.

> The SDK supports [OData](../odata/overview.mdx) and [OpenApi](../openapi/overview.mdx) services.

Reviewer: I'm voting to add `SDK` to Vale exclusions as it's very often a natural choice and makes docs more readable.

Reviewer: @florian-richter, you're the toughest guard of _The SAP Cloud SDK_. Do you think you can change your mind here? It's the most frequent Vale issue I see in PRs. I hope our readers know to what SDK we refer. Confusion is hardly possible and should be caught with proper reviews.

> For both service types the `execute()` methods triggers the request to the target system:

Reviewer: ```suggestion
For both service types the `execute()` methods triggers a request to a target system:
```

Reviewer: I think we're not implying specific requests and specific target systems. I'm not a guru of articles, it just feels we're talking here about a class of things, not specific things.

```ts
//OData
const responseOData = await BusinessPartner.requestBuilder()
  .getAll()
  .execute(myDestination);
//OpenApi
const responseOpenApi = await MyApi.myFunction().execute(myDestination);
```

`myDestination` is an SAP Cloud SDK representation of a <LinkToLatestJsApiDocs slug="interfaces/sap_cloud_sdk_core.Destination" name="destination" />.
The given request is executed against `myDestination`.

> For the rest of the document we will discuss the different options on destination lookup.

Reviewer: ```suggestion
For the rest of the document, we will discuss the different options on destination lookup.
```

> All statements apply to OData and OpenApi alike.

Reviewer: ```suggestion
All statements equally apply to OData and OpenAPI based services.
```
